Integrate SSD & EfficientDet tests, Add a dataloader and Deep Fish dataset processor

mohamedamrfaroukebrahim opened the pull request.

mohamedamrfaroukebrahim committed:
Increase test coverage by adding more unit tests for SSD300, SSD512, and EfficientDet.

mohamedamrfaroukebrahim committed:
Increase test coverage by adding more unit tests for SSD300, SSD512, and EfficientDet.

mohamedamrfaroukebrahim committed:
Remove redundant tests to eliminate duplication and improve maintainability

mohamedamrfaroukebrahim committed:
Rename efficientdet_all_test.py to efficientdet_test.py

mohamedamrfaroukebrahim committed:
Add a dataloader and Deep Fish dataset processor

- Added `from_yolo_dataloader.py` under `paz/backend` to convert any YOLO-format dataset to PAZ format.
- Added `Deepfish_data_processing.py` under `examples/fish_detection/` for downloading and processing the Deep Fish dataset into YOLO format.
- Moved dataset downloading to `paz/examples/fish_detection/Data/raw`.
- Stored processed dataset in `paz/examples/fish_detection/Data/processed`.
- Included `demo.py` to test all functionalities.

oarriaga commented on paz/models/detection/ssd512_test.py:
```diff
@@ -39,5 +39,55 @@ def test_incompatible_num_classes_with_weights():
+
+
 if __name__ == "__main__":
     pytest.main([__file__])
```
There is no need to add this line. Let''s just stick with the usual workflow of calling "pytest mytest.py" directly. Most likely adding this line could add unintended consequences if we start testing this way with "python mytest.py".

oarriaga commented on paz/models/detection/ssd300_test.py:
```diff
@@ -38,5 +38,61 @@ def test_incompatible_num_classes_with_weights():
+
+
 if __name__ == "__main__":
     pytest.main([__file__])
```
Yes as stated before let's remove this.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+import os
+import cv2
+import numpy as np
+from tqdm import tqdm
```
We can only import inside paz: numpy, jax, keras, cv2

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+from tqdm import tqdm
+
+# Define a tuple of valid image file extensions (lowercase)
+IMAGE_EXTENSIONS = (".jpg", ".jpeg", ".png", ".bmp", ".tif", ".tiff")
```
Let's avoid having global variables. Instead have a function that returns this extensions for example get_image_extensions

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+def validate_directories(images_directory: str, labels_directory: str):
+    """Validates that the provided directories exist."""
+    if not os.path.isdir(images_directory):
+        raise FileNotFoundError(
+            f"Images directory does not exist: {images_directory}"
+        )
```
Try to have full variable names but still have everything in a single line.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+        )
+
+
+def validate_file_correspondence(image_files: list, label_files: list):
```
this function is too long. Is it necessary to validate this? Are we not always sure that we are downloading the same dataset.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+        f
+        for f in os.listdir(images_directory)
+        if f.lower().endswith(IMAGE_EXTENSIONS)
+    ]
```
use always full variables names. If your python list comprehension does not fit in one line just use a for loop.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+        raise FileNotFoundError(
+            f"No image files found in the specified directory: {images_directory}"
+        )
+    return image_files
```
Make this a function.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+    return image_files
+
+
+def get_label_files(labels_directory: str):
```
same elements as the previous function.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+    Returns:
+        tuple: (width, height) of the image.
+    """
+    image = cv2.imread(image_path)
```
use internal functions instead i.e. paz.image.load

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+    with open(label_path, "r") as file:
+        for line in file:
+            # Convert each line to a list of floats
+            labels.append([float(x) for x in line.strip().split()])
```
This is hard to read split the list comprehension into another function.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+    return np.array(labels)
+
+
+def process_labels(
```
this function is too long. Let's split it into two one for each bool.

oarriaga commented on paz/backend/from_yolo_dataloader.py:
```diff
@@ -0,0 +1,215 @@
+    return np.array(processed_labels)
+
+
+def get_data_PAZ_formate(
```
I think formate is a typo.

oarriaga commented on examples/fish_detection/Deepfish_data_processing.py:
```diff
@@ -0,0 +1,226 @@
+import gdown
+
+
+def process_folder(src_folder, dst_img_folder, dst_label_folder):
```
this function is too long

oarriaga commented on examples/fish_detection/Deepfish_data_processing.py:
```diff
@@ -0,0 +1,226 @@
+    return gdrive_file_id
+
+
+def get_file(
```
this function is too long.

oarriaga commented on examples/fish_detection/Deepfish_data_processing.py:
```diff
@@ -0,0 +1,226 @@
+    output_path = os.path.join(output_dir, output_filename)
+
+    # Check if the URL is from Google Drive
+    if "drive.google.com" in urlparse(file_url).netloc:
```
you can split here in order to have a function that checks the validity of the link.

mohamedamrfaroukebrahim committed:
Address code requested changes.
- Remove __main__ blocks from SSD test files to align with pytest workflow
- Replace global IMAGE_EXTENSIONS with get_image_extensions() function
- Refactor long list comprehensions into explicit loops with full variable names
- Split validate_file_correspondence into modular validation functions
- Fix function name typo: get_data_PAZ_formate to get_data_PAZ_format
- Replace direct cv2.imread with paz.image.load for image loading
- Remove tqdm import and ensure compliance with library import policy

mohamedamrfaroukebrahim committed:
Handle not handled error of unavailable weights

oarriaga committed:
Merge branch 'paz-jax' into paz-jax
